compiler/ir.py

```diff
@@ -741,19 +742,19 @@ def apply_parallelization_to_node(self, node_id, parallelizer, fileIdGen, fan_ou
         elif splitter.is_splitter_consec_chunks():
             self.apply_consecutive_chunks_parallelization_to_node(node_id, parallelizer, fileIdGen, fan_out)
         else:
-            raise Exception("Splitter not yet implemented")
+            raise UnparallelizableError("Splitter not yet implemented")
 
     def apply_round_robin_parallelization_to_node(self, node_id, parallelizer, fileIdGen, fan_out,
                                                   r_split_batch_size):
         # TODO: this control flow should move done to aggregators once we implement them;
         #  currently, this cannot be done since splitter etc. would be added...
         aggregator_spec = parallelizer.get_aggregator_spec()
         if aggregator_spec.is_aggregator_spec_adj_lines_merge():
-            raise Exception("adj_lines_merge not yet implemented in PaSh")
+            raise UnparallelizableError("adj_lines_merge not yet implemented in PaSh")
```

Comment:
These 3 errors are actually general exceptions. Or rather NotImplemented errors, so I would make a new error type and catch them separately.